src/modules/75-ce/components/COInstanceSelector/COInstanceSelector.tsx

```diff
@@ -324,12 +358,55 @@ const InstancesFilter: React.FC<InstancesFilterProps> = ({
     handleGcpZonesFilterUpdate()
   }, [zonesData])
 
+  useEffect(() => {
+    if (tagsData?.response?.length) {
+      const keys = tagsData.response.map(tag => ({ label: tag.key as string, value: tag.key as string }))
+      setTagKeys(keys)
+    }
+  }, [tagsData])
+
+  useEffect(() => {
+    if (selectedTagPair?.key) {
+      const selectedTag = tagsData?.response?.find(tag => tag.key === selectedTagPair.key)
+      setTagValues(
+        _defaultTo(
+          selectedTag?.values?.map(v => ({ label: v, value: v })),
+          []
+        )
+      )
+    }
+  }, [selectedTagPair?.key])
```

Comment:
Codacy has a fix for the issue: [React Hook useEffect has a missing dependency: 'tagsData?.response'. Either include it or remove the dependency array.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9255099)
```suggestion
  }, [selectedTagPair.key, tagsData?.response])
```